File: src/java/SupportProof.java
import java.util.HashSet;
import java.util.Set;

public class SupportProof {
    
    static class PrimeIdeal {
        Set<String> elements;
        String name;
        
        PrimeIdeal(String name, String... elements) {
            this.name = name;
            this.elements = new HashSet<>();
            for (String e : elements) {
                this.elements.add(e);
            }
        }
        
        boolean contains(Set<String> ideal) {
            return elements.containsAll(ideal);
        }
    }
    
    public static void main(String[] args) {
        System.out.println("Support Module Proof - Java Implementation");
        System.out.println("Theorem: Supp(M) ⊆ V(Ann(M))\n");
        
        // Annihilator
        Set<String> annihilator = new HashSet<>();
        annihilator.add("x");
        annihilator.add("y");
        annihilator.add("z");
        
        System.out.println("Ann(M) = " + annihilator);
        
        // Prime ideals
        PrimeIdeal[] primes = {
            new PrimeIdeal("p1", "x", "y", "z", "a", "b"),
            new PrimeIdeal("p2", "x", "y", "a", "b"),
            new PrimeIdeal("p3", "x", "y", "z", "c", "d"),
            new PrimeIdeal("p4", "x", "a", "b", "c")
        };
        
        System.out.println("\nV(I) = {p ∈ Spec(A) | I ⊆ p}:");
        for (PrimeIdeal p : primes) {
            if (p.contains(annihilator)) {
                System.out.println("  " + p.name + " ∈ V(I)");
            }
        }
        
        System.out.println("\nMathematical Proof:");
        System.out.println("1. Take p ∈ Supp(M) (M_p ≠ 0)");
        System.out.println("2. Assume I ⊈ p → contradiction");
        System.out.println("3. Therefore I ⊆ p → p ∈ V(I)");
        System.out.println("4. Hence Supp(M) ⊆ V(I)");
        
        System.out.println("\n✓ Java implementation complete");
    }
}

File: src/go/support_proof.go
package main

import "fmt"

type PrimeIdeal struct {
	name     string
	elements map[string]bool
}

func NewPrimeIdeal(name string, elements []string) *PrimeIdeal {
	p := &PrimeIdeal{
		name:     name,
		elements: make(map[string]bool),
	}
	for _, e := range elements {
		p.elements[e] = true
	}
	return p
}

func (p *PrimeIdeal) Contains(ideal map[string]bool) bool {
	for e := range ideal {
		if !p.elements[e] {
			return false
		}
	}
	return true
}

func main() {
	fmt.Println("Support Module Proof - Go Implementation")
	fmt.Println("Theorem: Supp(M) ⊆ V(Ann(M))\n")
	
	// Annihilator
	annihilator := map[string]bool{
		"x": true,
		"y": true,
		"z": true,
	}
	
	fmt.Print("Ann(M) = {")
	for e := range annihilator {
		fmt.Printf("%s ", e)
	}
	fmt.Println("}\n")
	
	// Prime ideals
	primes := []*PrimeIdeal{
		NewPrimeIdeal("p1", []string{"x", "y", "z", "a", "b"}),
		NewPrimeIdeal("p2", []string{"x", "y", "a", "b"}),
		NewPrimeIdeal("p3", []string{"x", "y", "z", "c", "d"}),
		NewPrimeIdeal("p4", []string{"x", "a", "b", "c"}),
	}
	
	fmt.Println("V(I) contains:")
	for _, p := range primes {
		if p.Contains(annihilator) {
			fmt.Printf("  %s\n", p.name)
		}
	}
	
	fmt.Println("\nMathematical reasoning:")
	fmt.Println("Let p ∈ Supp(M), so M_p ≠ 0.")
	fmt.Println("Assume ∃a ∈ I with a ∉ p.")
	fmt.Println("Then a is unit in A_p, but aM = 0.")
	fmt.Println("Thus M_p = 0, contradiction.")
	fmt.Println("So I ⊆ p, hence p ∈ V(I).")
	
	fmt.Println("\n✓ Go implementation complete")
}
